File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Decorator/Java/SugarDecorator.java
public class SugarDecorator extends CoffeeDecorator {

    public double sugarCost = 0.5;

    public SugarDecorator(Coffee coffee) {
        super(coffee);
    }

    @Override
    public String getDescription() {
        return decoratedCoffee.getDescription() + ", Sugar";
    }

    @Override
    public double getCost() {
        return decoratedCoffee.getCost() + sugarCost;
    }
}

File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Decorator/Cpp/src/SugarDecorator.cpp
#include "../include/SugarDecorator.h"

SugarDecorator::SugarDecorator(Coffee* coffee) : CoffeeDecorator(coffee) {}

std::string SugarDecorator::getDescription() const {
    return decoratedCoffee->getDescription() + ", Sugar";
}

double SugarDecorator::getCost() const {
    return decoratedCoffee->getCost() + 0.5;
}
